### ISSUE TITLE
Crashes on launch after new iPadOS & app betas

### ISSUE THREAD

**Issue by erictbar** [croberts22/myanilist-bugs#32]
## Describe the issue

After TestFlight auto-installed the latest beta of MyAniList, opened app for first time in a couple days to a crash on splash screen. This was also my first launch of the app since the latest beta of iPadOS.

## How can you reproduce it?

1. Update to iPadOS 15.3 (19D504e) from previous beta
2. Allow TestFlight to download latest version of app
3. Open app

## What's the expected behavior?

App should open and start loading anime list

## How reproducible is this issue?
Also happened after reinstalling the app and restarting the iPad. However, after waiting a bit after restarting, was able to open the app.
Then, I tried to reinstall App Store version due to issue unable to restore premium features. Restored premium features, changed icon and theme. Then installed TestFlight beta over. App crashed again. Waited a couple minutes and the app started correctly.

## Device Information

iOS Version: iPadOS 15.3 (19D5040e) beta

MyAniList Version: 2.10.2 (97e6e85c)

iOS Device: iPad Pro 10.5"

Jailbroken: No

### Additional Context

3 crash reports hopefully sent via Apple TestFlight UI

**Comment by seanblue** [comment]
The iPhone version crashed for me the first time I opened it after the most recent update. I thought it was a coincidence, but with you posting this issue I guess it's a real bug.

**Comment by croberts22** [comment]
This may be a two-fold issue, but I was able to identify one of the root causes here. It looks like there might actually be an issue with the [Swift compiler in the most recent version of their tools.](https://developer.apple.com/forums/thread/697070) Luckily there's a workaround (albeit inconvenient) and this will be included in the next beta.

I'll check those reports and see if there are any similarities in those crashlogs versus the ones I've been seeing that relate to the compiler issue. Thanks for sending those in!

**Comment by erictbar** [comment]
The issue is occurring again after opening the app to the new TestFlight beta (2c8ab3e8)
I also noticed that when the app started, it asked for Touch ID (which I have enabled) alongside (perhaps before) the update screen. 
After dismissing the update screen, I had to press the unlock button a couple of times before it would bring up Touch ID again.
Then my anime list launched, and the lag came back for about 10 seconds, even though the list wasn't being updated. 
When refreshing my list the lag also occurred, and that replicated to the same degree as my original post.
